tests/st/utils/utils.py

```diff
@@ -147,6 +147,22 @@ def assert_no_error(self, text=None):
         if text:
             self.assert_output_contains(text)
 
+    def check_reports(self, reports):
```

Comment:
I wasn't sure why this is part of the CalicoctlOutput class - but then it looks like we're using the CalicoctlOutput class for both `calicoctl` commands and `calico-upgrade` which doesn't feel correct to me.  I suspect we should have had a separate CalicoUpgradeOutput class that give you a nice way to output additional diags if any of the assertions are invalid - e.g. assert_no_error could also output the contents of all of the reports if there was an error.